httomo/cli.py

```diff
@@ -190,21 +164,50 @@ def run(
     tr = TransformLayer(comm=comm, save_all=save_all)
     pipeline = tr.transform(pipeline)
 
-    # Run the pipeline using Taskrunner, with temp dir or reslice dir
-    mon = make_monitors(monitor)
-    ctx: AbstractContextManager = nullcontext(reslice_dir)
-    if reslice_dir is None:
-        ctx = tempfile.TemporaryDirectory()
-    with ctx as tmp_dir:
-        runner = TaskRunner(
-            pipeline,
-            Path(tmp_dir),
-            monitor=mon,
-            memory_limit_bytes=memory_limit,
-        )
-        runner.execute()
-        if mon is not None:
-            mon.write_results(monitor_output)
+    if not does_contain_sweep:
+        # we use half the memory for blocks since we typically have inputs/output
+        memory_limit = transform_limit_str_to_bytes(max_memory) // 2
+
+        if max_cpu_slices < 1:
+            raise ValueError("max-cpu-slices must be greater or equal to 1")
+        httomo.globals.MAX_CPU_SLICES = max_cpu_slices
+
+        # try to access the GPU with the ID given
+        try:
+            import cupy as cp
+
+            gpu_count = cp.cuda.runtime.getDeviceCount()
+
+            if gpu_id != -1:
+                if gpu_id not in range(0, gpu_count):
+                    raise ValueError(
+                        f"GPU Device not available for access. Use a GPU ID in the range: 0 to {gpu_count} (exclusive)"
+                    )
+
+                cp.cuda.Device(gpu_id).use()
+
+            httomo.globals.gpu_id = gpu_id
+
+        except ImportError:
+            pass  # silently pass and run if the CPU pipeline is given
```

Comment:
Can we factor this out into a helper function? It's a bit distracting here in this function.

Reply:
Good idea, I agree that this block of code is distracting being in the top-level function of the CLI.

Reply:
Done in c4ca5ff